apps/aechannel/src/aesc_close_mutual_tx.erl

```diff
@@ -69,134 +72,194 @@ fee(#channel_close_mutual_tx{fee = Fee}) ->
-    InitiatorPubKey.
+-spec origin(tx()) -> pubkey() | undefined.
+origin(#channel_close_mutual_tx{channel_id = ChannelId}) ->
+    case aec_chain:get_channel(ChannelId) of
+        {ok, Channel} ->
+            aesc_channels:initiator(Channel);
+        {error, not_found} -> undefined
+    end.
 
 -spec check(tx(), aec_trees:trees(), height()) -> {ok, aec_trees:trees()} | {error, term()}.
-check(#channel_close_mutual_tx{channel_id  = ChannelId,
-                               initiator   = InitiatorPubKey,
-                               participant = ParticipantPubKey,
-                               amount      = Amount,
-                               fee         = Fee,
-                               nonce       = Nonce}, Trees, Height) ->
-    Checks =
-        [fun() -> aetx_utils:check_account(InitiatorPubKey, Trees, Height, Nonce, Fee) end,
-         fun() -> aesc_utils:check_active_channel_exists(ChannelId, InitiatorPubKey, ParticipantPubKey, Trees) end,
-         fun() -> check_peer_has_funds(InitiatorPubKey, ParticipantPubKey, ChannelId, Amount, Trees) end],
-    case aeu_validation:run(Checks) of
-        ok ->
-            {ok, Trees};
-        {error, _Reason} = Error ->
-            Error
+check(#channel_close_mutual_tx{channel_id       = ChannelId,
+                               from             = _From,
+                               initiator_amount = InitiatorAmount,
+                               responder_amount = ResponderAmount,
+                               ttl              = TTL,
+                               fee              = Fee,
+                               nonce            = Nonce}, Trees, Height) ->
+    case aesc_state_tree:lookup(ChannelId, aec_trees:channels(Trees)) of
+        none ->
+            {error, channel_does_not_exist};
+        {value, Channel} ->
+            InitiatorPubKey = aesc_channels:initiator(Channel),
+            Checks =
+                [fun() -> aetx_utils:check_account(InitiatorPubKey, Trees, Height, Nonce, Fee) end,
+                 fun() ->
+                    case aesc_channels:is_active(Channel) of
+                        true -> ok;
+                        false -> {error, channel_not_active}
+                    end
+                end,
+                fun() -> % check fee
+                    ChannelAmt = aesc_channels:initiator_amount(Channel) +
+                                 aesc_channels:participant_amount(Channel),
+                    ok_or_error(ChannelAmt >= Fee, fee_too_big)
+                end,
+                fun() -> % check amounts
+                    ChannelAmt = aesc_channels:initiator_amount(Channel) +
+                                 aesc_channels:participant_amount(Channel),
+                    ok_or_error(ChannelAmt =:= InitiatorAmount + ResponderAmount + Fee,
+                                wrong_amounts)
+                end,
+                fun() -> % check TTL
+                    ok_or_error(TTL >= Height, ttl_expired)
```

Comment:
Are we sure this is what TTL is meant to do? And Isn't that standard transaction TTL that is supposed to be implemented for all blockchain transaction? Then shouldn't that value be just a positive number and be relative to current Height? 

In specs I can see only
```
Peers will want to set the ttl to a value greater than zero, to avoid uncertainty. If the fees included are low and transaction pressure is high, then the transaction might end up being stuck in the mempool for an extended period.
```

In oracles/naming service TTL was specific to these services, and described for how long oracle query/response/name/whatnot will live. Standard transaction TTL was not introduced there yet (AFAIK).

Reply:
I was pointed to [This pivotal story](https://www.pivotaltracker.com/n/projects/2124891/stories/153897559)

Reply:
Good 👍